cli/internal/experiment/experiment.go

```diff
@@ -0,0 +1,25 @@
+package experiment
+
+import (
+	"os"
+	"strings"
+)
+
+type Experiment string
+
+const (
+	// DependencyInjection enables the dependency injection experiment,
+	// generating code to facilitate dependency injection.
+	DependencyInjection Experiment = "di"
+)
+
+// Enabled reports whether the given experiment is enabled.
+func Enabled(x Experiment) bool {
+	fields := strings.Fields(os.Getenv("ENCORE_EXPERIMENT"))
```

Comment:
> It’s not yet something that you should be able to opt in to on an app level,

yet it will impact at a per app level - if two people are working on an app and one of them starts using DI, then the other needs to turn the feature on in the deamon. Where as could track which apps have it switched on and have the daemon only operate on those apps

Reply:
i.e. it the behaviour changes should not impact apps not using the feature - so if the daemon's startup behaviour is impacted - can that be refactored to only impact opt'ed in apps?

Reply:
(Also does this mean that the platform _will not support_ DI? And anybody using this feature during the experimental phase therefore cannot deploy to the cloud?